src/plugins/vis_type_timelion/server/usage_collector/get_usage_collector.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import moment from 'moment';
+import { SearchResponse } from 'elasticsearch';
+import { ElasticsearchClient } from 'src/core/server';
+import { SavedObjectsClientContract, ISavedObjectsRepository } from 'kibana/server';
+import { IFieldType } from '../../../data/common';
+import { getIndexArgs } from '../../common/parser';
+
+type ESResponse = SearchResponse<{ visualization: { visState: string }; updated_at: string }>;
+
+interface IndexPatternSavedObjectAttrs {
+  title: string;
+  fields: string;
+}
+
+export interface TimelionUsage {
+  timelion_use_scripted_fields_90_days_total: number;
+}
+
+const getPastDays = (dateString: string): number => {
+  const date = moment(dateString);
+  return moment().diff(date, 'days');
+};
+
+export const getStats = async (
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract | ISavedObjectsRepository,
+  index: string
+): Promise<TimelionUsage | undefined> => {
+  const indexPatterns = await soClient.find<IndexPatternSavedObjectAttrs>({
```

Comment:
@stratula have a question cause that code request data for each available index patterns in instance. Probably it's ok, but as alternative solution we can request index pattern objects separately. Two solutions have own pros/cons. I just want to think about it in prism of performance degradation.

Reply:
I agree, now I request index Pattern for each timelion expression separately.